extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java

```diff
@@ -151,10 +153,16 @@ public void register(RegistrationContext context) {
                 for (RootDefinition rootDefinition : configItem.getReadResult().getAllRoots()) {
                     if (rootDefinition.getConfigPhase() == ConfigPhase.BUILD_AND_RUN_TIME_FIXED
                             || rootDefinition.getConfigPhase() == ConfigPhase.RUN_TIME) {
-                        context.configure(rootDefinition.getConfigurationClass()).types(rootDefinition.getConfigurationClass())
+                        Class<?> configRootClass = rootDefinition.getConfigurationClass();
+                        context.configure(configRootClass).types(configRootClass)
                                 .scope(Dependent.class).creator(mc -> {
                                     // e.g. return Config.ApplicationConfig
-                                    mc.returnValue(mc.readStaticField(rootDefinition.getDescriptor()));
+                                    ResultHandle configRoot = mc.readStaticField(rootDefinition.getDescriptor());
+                                    mc.ifNull(configRoot).trueBranch().throwException(CreationException.class,
+                                            String.format(
+                                                    "Config root %s [%s] was not initialized yet. Make sure the bean is accessed at the right time in the correct bootstrap phase.",
```

Comment:
Perhaps we should simplify the message to just say that config roots can only be accessed at runtime init? Mentioning bootstrap here is kind of confusing since there is also the bootstrap config phase.

Reply:
> Perhaps we should simplify the message to just say that config roots can only be accessed at runtime init

Well, `BUILD_AND_RUN_TIME_FIXED` roots can be safely used in STATIC_INIT.

> Mentioning bootstrap here is kind of confusing since there is also the bootstrap config phase.

Hm, so it's bootstrap phase VS bootstrap config phase. I'm not sure about the correct wording.

Reply:
Yeah it's tricky. Maybe let's just go with what you added and see if people get confused

Reply:
I'll remove the next sentence...

Reply:
Done.

Reply:
:+1: